### ISSUE TITLE
Field consistency

### ISSUE THREAD

**Issue by DavidBrainard** [isetbio/isetbio#20]
I fear we need a really careful review of object precedence and field consistency.

I discovered that sensorSet('cone type') didn't update the cfa.pattern field, so that two places where the same information could be stored in the sensor structure could get out of whack with each other.  This caused some surprising results.

I updated sensorSet('cone type') so that it also updates the cfa pattern.  I"m viewing this as a quick patch.  We will run into the same problem if one sets the cfa or cfa pattern -- I didn't fix it in that case because there won't always be a cone type field and I am charging on with other things.

My sense is that we shouldn't really have both a cone type and cfa.pattern field in the sensor struct, and that setting/getting cone type should just operate as synonyms for setting/getting the cfa pattern.  But there may be reasons the second field crept in.

More generally, as structures get set as part of other structures more and more (cone, pixel, wvf, etc.), who is in charge of redundant parameters probably needs to be carefully specified and implemented.

**Comment by chichilnisky** [comment]
This sounds pretty important. Ideally one would have no redundant parameters at all, as a strict and important design choice. One exception might be caching for speed, which is perhaps the reason here. 

ej

**Comment by wandell** [comment]
David is right.  And in fact HJ and I are in the middle of a related issue, with a similar structure.

In building up the sensor structure to handle human parameters, I started with the ISET camera sensor.  There are a number of features, however, that are present in humans and not present in cameras.  This includes things such as the inert pigments (lens, macular) and also rather extreme inhomogeneity of the receptor mosaic with eccentricity (called field height in sensor land).

HJ managed the wide array of additional parameters by adding (in ISETBIO) the .human slot into the sensor.  And then he put the .cone slot into .human.  Adding these structures was both essential and also potentially conflicting.  David is pointing out a case of conflict.  But parameters like the cone photopigment density and related parameters needed a home, and HJ chose the cone type.

HJ and I were trying to smooth over some of the issues between other parameterizations of the sensor based on the original ISET code and the new parametrization based on the data in the .human.cone slot.  We made a branch to work on this and test it, but haven't alerted others yet.

The issue David raises is in a similar conceptual space.  As we enrich the data structure to handle human parameters, there can be either conflicts or at least overloading of the parameters in the original ISET parameterization.  Checking with one another to make sure that we aren't introducing conflicts, and doing more unit testing, is something I know we all agree on.   

The 'cone type' case above is an example in which there is a duplicate representation.  I am guessing we all think there should only be one place.  Let's get agreement on this one - and perhaps others as they arise - in our phone conversations.
